### PR TITLE
Export submit-addon via package.json

### LINKED ISSUE #3241: Allow programatically signing from an xpi file
### Is this a feature request or a bug?
Feature request

### What is the current behavior?

Programmatically, the only way to sign is using `webExt.cmd.sign`, which also does a build. https://github.com/mozilla/sign-addon is deprecated, likely because it was included in utils/submit-addon.js. There doesn't seem to be a way to trigger a submission for an existing xpi.

### What is the expected or desired behavior?

If we add utils/submit-addon to exports, I'd be able to use it directly. I'm open to other ideas.

### PR DESCRIPTION
**Pull request by kewisch**
Fixes #3241

### CODE REVIEW

**Comment by kewisch** [comment]
Test failure is blocking sec issues, I didn't add any deps so I don't think I can fix this.

Note that locally the test I added runs as "pending", both with and without my changes. Might be something up with that one?

**Comment by Rob--W** [comment]
The npm audit issues are pre-existing. I think that @rpl or @willdurand was going to address that.

**Comment by rpl** [comment]
> The npm audit issues are pre-existing. I think that @rpl or @willdurand was going to address that.

yes, I was the one that was meant to followup on that.

The transitive dependency that was making us hitting the npm audit failure is part of sinon dependencies and despite the fact that the previous PR that was updating sinon to 18.0.1 was still failing (which was what I was going to take a look into), the new PR for sinon 19.0.0 (#3251) did actually pass the npm audit CI job step and so I proceeded with merging it.

@kewisch would you mind to rebase this PR to confirm that the CI job failure is gone as expected?

**Comment by kewisch** [comment]
Looks like we're good :)

**Comment by rpl** [comment]
We discussed about this in the triage meeting and we agreed to move forward with merging this PR but also update the README.md file to clarify what `Note: There is limited support for this API.` means (in particular that there is no guarantees about backward compatibility in new web-ext versions).

**Comment by kewisch** [comment]
I added a note to the readme, does that look good?

**Comment by rpl** [comment]
> I added a note to the readme, does that look good?

In comment https://github.com/mozilla/web-ext/pull/3250#issuecomment-2462207299 I was referring to expanding/rewording the existing `**Note:** There is limited support for this API` that we have at the start of the section here: https://github.com/mozilla/web-ext/blob/cde10a871e60f1fede2a79539563b13122f299bf/README.md?plain=1#L126

Nonetheless, I think the note you added may also be worth keeping (because the sign command entrypoint is more likely to stay stable then the internal signAddon module).

In the meantime, I'm going to approving and merging this version, we can followup to tweak that other note in a separate PR.

File: README.md [hunk 1]
```diff
@@ -194,6 +194,24 @@ webExtLogger.consoleStream.makeVerbose();
...
+
+signAddon({
+  apiKey,
+  apiSecret,
+  amoBaseUrl: 'https://addons.mozilla.org/api/v5/',
+  id: 'extension-id@example.com',
+  xpiPath: pathToExtension,
+  savedUploadUuidPath: '.amo-upload-uuid',
+  channel: 'unlisted',
+  userAgentString: 'web-ext/0.2',
```
[the remark this answers is not in the document]
Comment (rpl) [reply README.md:211]: @kewish Apologies for the long time needed for us to come back to this PR and merge it. I've finally merged this PR today and created #3327 to followup with this kind of tweak (and #3326 to set a fallback userAgentString if it is completely omitted).

**Pull request merged**